apps/web/src/app/vault/components/vault-items/vault-collection-row.component.ts

```diff
@@ -26,13 +34,28 @@ export class VaultCollectionRowComponent {
   @Input() canDeleteCollection: boolean;
   @Input() organizations: Organization[];
   @Input() groups: GroupView[];
+  @Input() orgVault: boolean;
+  @Input() flexibleCollectionsEnabled: boolean;
+  permissionText: string;
 
   @Output() onEvent = new EventEmitter<VaultItemEvent>();
 
   @Input() checked: boolean;
   @Output() checkedToggled = new EventEmitter<void>();
 
-  constructor(private router: Router, private activatedRoute: ActivatedRoute) {}
+  constructor(private router: Router) {}
+
+  ngOnInit() {
+    if (this.collection.manage) {
+      this.permissionText = "canManage";
+    } else if (!this.collection.readOnly) {
+      this.permissionText = "canEdit";
+    } else if (this.collection.readOnly) {
+      this.permissionText = "canView";
+    } else {
+      this.permissionText = null;
+    }
```

Comment:
You could use this to set the fallback value instead of checking for `!permissionText` in the template. i.e.

```suggestion
  ngOnInit() {
    if (this.collection.manage) {
      this.permissionText = this.i18nService.t("canManage");
    } else if (!this.collection.readOnly) {
      this.permissionText = this.i18nService.t("canEdit");
    } else if (this.collection.readOnly) {
      this.permissionText = this.i18nService.t("canView");
    } else {
      this.permissionText = "-";
    }
```

Also note this is not reactive, i.e. it will not change if the collection is updated, because it's only calculated `onInit`. I suggest making it a getter or using a setter on the `Input`, or you could even move this logic to the collection object itself.